### PR TITLE
Update logrus to v1.0.1

### PR DESCRIPTION
**Pull request by dmcgowan**
Updates logrus to v1.0.1 and fixes the vendor case sensitivity issue. This issue is being fixed in parallel with many other projects to prevent future vendoring errors.

See https://github.com/sirupsen/logrus#logrus-

### CODE REVIEW

**Comment by mlaventure** [comment]
@dmcgowan needs a little rebase :)

**Comment by fcrisciani** [comment]
@dmcgowan if you can rebase and solve the conflict we can have this PR get in

**Comment by mavenugo** [comment]
@dmcgowan this is blocking the upcoming libnetwork vendoring due to the change (https://github.com/moby/moby/pull/34272) which was merged.

**Comment by dmcgowan** [comment]
Pushed an update. This change is a lot bigger than it should be. The vendor was pretty out of sync compared to what is used by moby and swarmkit. I have updated the needed packages to match. If the changes cause issues, that is another issue since upgrading this vendor downstream will be using these versions....

**Comment by fcrisciani** [comment]
@dmcgowan looks like there is still missing dependencies PTAL

**Comment by dmcgowan** [comment]
Update, not sure how latest CI error would be related

I changed the vendors to exclude any potential functional changes in the update. I would encourage the maintainers of this project to update the vendored versions. As is, the out of date vendors makes importing libnetwork riskier as it has not been tested with the versions of the packages it is likely to be used with in Docker and SwarmKit.

**Comment by fcrisciani** [comment]
LGTM

File: vendor.conf [hunk 1]
```diff
@@ -31,9 +30,10 @@ github.com/hashicorp/go-sockaddr acd314c5781ea706c710d9ea70069fd2e110d61d
 github.com/hashicorp/serf 598c54895cc5a7b1a24a398d635e8c0ea0959870
 github.com/mattn/go-shellwords 525bedee691b5a8df547cb5cf9f86b7fb1883e24
 github.com/miekg/dns d27455715200c7d3e321a1e5cadb27c9ee0b0f02
-github.com/opencontainers/runc ba1568de399395774ad84c2ace65937814c542ed
+github.com/opencontainers/runc 8694d576ea3ce3c9e2c804b7f91b4e1e9a575d1c https://github.com/dmcgowan/runc.git
```
Comment (fcrisciani) [inline vendor.conf]: Are these still to be migrated?

File: vendor.conf [hunk 2]
```diff
@@ -11,9 +10,9 @@ github.com/coreos/etcd 925d1d74cec8c3b169c52fd4b2dc234a35934fce
 github.com/coreos/go-systemd b4a58d95188dd092ae20072bac14cece0e67c388
 github.com/deckarep/golang-set ef32fa3046d9f249d399f98ebaf9be944430fd1d
 
-github.com/docker/docker 9c96768eae4b3a65147b47a55c850c103ab8972d
-github.com/docker/go-connections 34b5052da6b11e27f5f2e357b38b571ddddd3928
-github.com/docker/go-events 2e7d352816128aa84f4d29b2a21d400133701a0d
+github.com/docker/docker 2cac43e3573893cf8fd816e0ad5615426acb87f4 https://github.com/dmcgowan/docker.git
```
Comment (fcrisciani) [inline vendor.conf]: Are these still to be migrated?

File: vendor.conf (hunk 1 above)
Comment (fcrisciani) [inline vendor.conf]: From moby/moby the git commit is:
`github.com/opencontainers/runc e9325d442f5979c4f79bfa9e09bdf7abb74ba03b https://github.com/dmcgowan/runc.git
`
Should we keep this aligned?

File: vendor.conf (hunk 2 above)
Comment (fcrisciani) [inline vendor.conf]: can this one be the master one now that the logrus changes are in already there?

**Comment by fcrisciani** [comment]
Tried to vendor the same runc of moby and the latest moby master but hit some extra dependency issue:

```
go build -tags experimental -o "bin/dnet" ./cmd/dnet                
vendor/github.com/docker/docker/daemon/cluster/convert/config.go:5:2: cannot find package "github.com/docker/swarmkit/api" in any of:   
        /go/src/github.com/docker/libnetwork/vendor/github.com/docker/swarmkit/api (vendor tree)                                        
        /usr/local/go/src/github.com/docker/swarmkit/api (from $GOROOT)                                                                 
        /go/src/github.com/docker/swarmkit/api (from $GOPATH)       
vendor/github.com/docker/docker/daemon/cluster/convert/service.go:11:2: cannot find package "github.com/docker/swarmkit/api/genericresource" in any of:                                                                                                                         
        /go/src/github.com/docker/libnetwork/vendor/github.com/docker/swarmkit/api/genericresource (vendor tree)                        
        /usr/local/go/src/github.com/docker/swarmkit/api/genericresource (from $GOROOT)                                                 
        /go/src/github.com/docker/swarmkit/api/genericresource (from $GOPATH)                                                           
vendor/github.com/docker/docker/daemon/cluster/convert/swarm.go:10:2: cannot find package "github.com/docker/swarmkit/ca" in any of:    
        /go/src/github.com/docker/libnetwork/vendor/github.com/docker/swarmkit/ca (vendor tree)                                         
        /usr/local/go/src/github.com/docker/swarmkit/ca (from $GOROOT)                                                                  
        /go/src/github.com/docker/swarmkit/ca (from $GOPATH)        
vendor/github.com/docker/docker/daemon/cluster/convert/config.go:6:2: cannot find package "github.com/gogo/protobuf/types" in any of:   
        /go/src/github.com/docker/libnetwork/vendor/github.com/gogo/protobuf/types (vendor tree)                                        
        /usr/local/go/src/github.com/gogo/protobuf/types (from $GOROOT)                                                                 
        /go/src/github.com/gogo/protobuf/types (from $GOPATH)       
vendor/github.com/docker/docker/api/types/registry/registry.go:7:2: cannot find package "github.com/opencontainers/image-spec/specs-go/v1" in any of:                                                                                                                           
        /go/src/github.com/docker/libnetwork/vendor/github.com/opencontainers/image-spec/specs-go/v1 (vendor tree)                      
        /usr/local/go/src/github.com/opencontainers/image-spec/specs-go/v1 (from $GOROOT)                                               
        /go/src/github.com/opencontainers/image-spec/specs-go/v1 (from $GOPATH)                                                         
vendor/github.com/opencontainers/runc/libcontainer/configs/config.go:10:2: cannot find package "github.com/opencontainers/runtime-spec/specs-go" in any of:                                                                                                                     
        /go/src/github.com/docker/libnetwork/vendor/github.com/opencontainers/runtime-spec/specs-go (vendor tree)                       
        /usr/local/go/src/github.com/opencontainers/runtime-spec/specs-go (from $GOROOT)                                                
        /go/src/github.com/opencontainers/runtime-spec/specs-go (from $GOPATH)
```

**Comment by mavenugo** [comment]
Thanks @fcrisciani .... Lets use https://github.com/docker/libnetwork/issues/1899 to track this change and get this PR merged asap so that vendoring can happen in moby.

**Pull request merged**